go/backend/stock/stock.go

```diff
@@ -21,6 +22,20 @@ import (
 
 //go:generate mockgen -source stock.go -destination stock_mocks.go -package stock -exclude_interfaces Index
 
+// ReadOnly is a read-only interface to a Stock.
+// It allows for retrieving values associated with index values,
+// while it can neither create nor delete values.
+type ReadOnly[I Index, V any] interface {
+	// Get retrieves a value associated to an index value. If there is no such
+	// element, an undefined value is returned. Stock implementations are not
+	// required to retain information on valid and deleted indexes to
+	// efficiently distinguish valid from invalid accesses. The life-cycle of
+	// IDs is expected to be managed by the client code.
+	Get(I) (V, error)
```

Comment:
nit: the documentation string is matching the `Stock` use case and would need to be adjusted here and there for the `ReadOnly` stock. It might be simpler to not do the embedding and define an independent interface.

Reply:
ok, done this way. I basically removed the second part of the documentation, which was unrelated.